src/hotspot/share/gc/g1/g1ConcurrentMark.cpp

```diff
@@ -1440,6 +1440,12 @@ void G1ConcurrentMark::remark() {
       _needs_remembered_set_rebuild = (cl.total_selected_for_rebuild() > 0);
     }
 
+    if (_needs_remembered_set_rebuild) {
+      // Prune rebuild candidates based on G1HeapWastePercent.
+      // Improves rebuild time in addition to remembered set memory usage.
+      G1CollectionSetChooser::build(_g1h->workers(), _g1h->num_regions(), _g1h->policy()->candidates());
+    }
+
```

Comment:
I would prefer to put this block into above block so that it is part of the "Select For Rebuild and Reclaim Empty Regions" measurement.